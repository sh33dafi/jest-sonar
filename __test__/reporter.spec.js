const Reporter = require('../src/reporter');

describe('reporter', () => {
    let reporter;

    beforeEach(() => {
        reporter = new Reporter('/the/root');
    });

    describe('When calling toSonarReport', () => {
        test('should map a test result file to a sonar report', () => {
            const testResults = {
                testResults: [
                    {
                        testFilePath: '/the/root/my-test/my-test.spec.js',
                        testResults: [
                            {
                                fullName: 'When doing this should "be" ok',
                                duration: 30,
                                failureMessages: []
                            },
                            {
                                fullName:
                                    'When this is failing the sonar report should contain the failure',
                                duration: 10,
                                failureMessages: ['A failure']
                            }
                        ]
                    },
                    {
                        testFilePath: '/the/root/my-test/my-test.spec.js',
                        testResults: [
                            {
                                fullName: `Some crazy title with ${String.fromCharCode(
                                    27
                                )}[31;1m red color ${String.fromCharCode(
                                    27
                                )}[0m and ðŸ”¥`,
                                duration: 30,
                                failureMessages: []
                            },
                            {
                                fullName:
                                    'When this is failing the sonar report should contain the failure',
                                duration: 10,
                                failureMessages: ['A failure']
                            }
                        ]
                    },
                    {
                        testFilePath: '/the/root/my-second-test.spec.js',
                        testResults: [
                            {
                                fullName: 'When doing this should be ok',
                                duration: 35,
                                failureMessages: []
                            }
                        ]
                    },
                    {
                        testFilePath: '/the/root/my-skipped-test.spec.js',
                        testResults: [
                            {
                                fullName: 'Skipped',
                                duration: 10,
                                failureMessages: [],
                                status: 'pending'
                            }
                        ]
                    }
                ]
            };

            expect(reporter.toSonarReport(testResults)).toMatchSnapshot();
        });
    });
});
