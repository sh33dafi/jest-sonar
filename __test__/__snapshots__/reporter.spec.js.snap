// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reporter When calling toSonarReport should map a test result file to a sonar report 1`] = `
"<testExecutions version=\\"1\\">
<file path=\\"my-test/my-test.spec.js\\">
<testCase name=\\"When doing this should &quot;be&quot; ok\\" duration=\\"30\\" />
<testCase name=\\"When this is failing the sonar report should contain the failure\\" duration=\\"10\\">
<failure message=\\"Error\\"><![CDATA[A failure]]></failure>
</testCase>
</file>
<file path=\\"my-test/my-test.spec.js\\">
<testCase name=\\"Some crazy title with  red color  and &#x1F525;\\" duration=\\"30\\" />
<testCase name=\\"When this is failing the sonar report should contain the failure\\" duration=\\"10\\">
<failure message=\\"Error\\"><![CDATA[A failure]]></failure>
</testCase>
</file>
<file path=\\"my-second-test.spec.js\\">
<testCase name=\\"When doing this should be ok\\" duration=\\"35\\" />
</file>
<file path=\\"my-skipped-test.spec.js\\">
<testCase name=\\"Skipped\\" duration=\\"10\\">
<skipped message=\\"Skipped\\"/>
</testCase>
</file>
</testExecutions>"
`;
